Dockerfile

```diff
@@ -1,26 +1,20 @@
-FROM golang:alpine AS build
+FROM golang:1.14.2-alpine AS builder
 WORKDIR /app
-ADD . /app
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o ./bin/ingress-controller ./cmd/caddy
 
-FROM alpine:latest as certs
-RUN apk --update add ca-certificates
-
-FROM golang:1.13.5 as builder
-WORKDIR /build
-ENV CGO_ENABLED=0
-ENV GOOS=linux
-ENV GOARCH=amd64
-RUN mkdir -p ./bin
 COPY go.mod go.sum ./
 RUN go mod download
+
 COPY ./cmd ./cmd
 COPY ./pkg ./pkg
 COPY ./internal ./internal
-RUN go build -o ./bin/ingress-controller ./cmd/caddy
 
-FROM scratch
-COPY --from=builder /build/bin/ingress-controller .
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o ./bin/ingress-controller ./cmd/caddy
+
+FROM alpine:latest as certs
+RUN apk --update add ca-certificates
+
+FROM alpine:latest
```

Comment:
WDYT about using `scratch` image here? Caddy is using the scratch base image for the official distribution [1], so I'll suggest to keep the image small as possible :)

[1] https://hub.docker.com/r/caddy/caddy/dockerfile

Reply:
Yep good catch, I replaced it for debug purpose 👍